container_pipeline/workers/delivery.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python
+
+import hashlib
+import json
+import logging
+import os
+
+from container_pipeline.lib.log import load_logger
+from container_pipeline.lib.openshift import Openshift, OpenshiftError
+from container_pipeline.workers.base import BaseWorker
+from container_pipeline.utils import get_job_name, get_project_name
+
+
+class DeliveryWorker(BaseWorker):
+    """
+    Delivery Worker tags the image built by Build Worker using the
+    `desired-tag` field in index entry
+    """
+    def __init__(self, logger=None, sub=None, pub=None):
+        super(DeliveryWorker, self).__init__(logger, sub, pub)
+        self.openshift = Openshift(logger=self.logger)
+
+    def handle_job(self, job):
+        """Handles a job meant for delivery worker"""
+        # TODO: this needs to be addressed after addressing CentOS#278
+        self.logger.info('Starting delivery for job: {}'.format(job))
+
+        success = self.deliver_build(job)
+
+        if success:
+            self.handle_delivery_success(job)
+        else:
+            self.handle_delivery_failure(job)
+
+    def deliver_build(self, job):
+        """
+        Runs an `oc build` with the `run_delivery.sh` script as a part of build
+        template. It mainly changes the tag of the image from a test tag
+        generated by build process to the tag desired by user as mentioned in
+        `desired-tag` field in cccp.yml
+        """
+        namespace = get_job_name(job)
+        project = hashlib.sha224(namespace).hexdigest()
```

Comment:
can't we put this hash into the get_job_name function itself? as that would be single point of change if we want to change that later.